lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_static_class/visitor.dart

```diff
@@ -0,0 +1,52 @@
+part of 'prefer_static_class_rule.dart';
+
+class _Visitor extends RecursiveAstVisitor<void> {
+  _Visitor({
+    required bool ignorePrivate,
+    required Iterable<String> ignoreNames,
+    required Iterable<String> ignoreAnnotations,
+  })  : _ignorePrivate = ignorePrivate,
+        _ignoreNames = ignoreNames.map(RegExp.new),
+        _ignoreAnnotations = ignoreAnnotations;
+
+  final bool _ignorePrivate;
+  final Iterable<RegExp> _ignoreNames;
+  final Iterable<String> _ignoreAnnotations;
+
+  final _declarations = <AstNode>[];
+
+  Iterable<AstNode> get declarations => _declarations;
+
+  @override
+  void visitTopLevelVariableDeclaration(TopLevelVariableDeclaration node) {
+    super.visitTopLevelVariableDeclaration(node);
+
+    if (!_hasIgnoredAnnotation(node) &&
+        !node.variables.variables
+            .every((variable) => _shouldIgnoreName(variable.name.lexeme))) {
+      _declarations.add(node);
+    }
+  }
+
+  @override
+  void visitFunctionDeclaration(FunctionDeclaration node) {
+    super.visitFunctionDeclaration(node);
+    final nodeName = node.name.lexeme;
+    if (node.parent is CompilationUnit &&
+        !_hasIgnoredAnnotation(node) &&
+        !_shouldIgnoreName(nodeName) &&
+        nodeName != 'main') {
```

Comment:
I believe `isEnitrypoint` helper function should be used here. It covers not only `main`, but also other possible entry points.